### PR TITLE
add GCC9 support, update range-v3 to 1.0-beta

### PR DESCRIPTION
**Pull request by h-2**
Please review, but do not merge, yet. I recommend reviewing commit-by-commit.

We need to have the blockers fixed, before we can merge this (everything builds locally with applied patches in range-v3 and sdsl).

Blockers (GCC7 and GCC8):
  * ~~ericniebler/range-v3#1023~~
  * ~~xxsds/sdsl-lite#57~~
  * ~~xxsds/sdsl-lite#58~~

Blockers (GCC9, fixes available in next snapshot):
  * ~~ericniebler/range-v3#1011~~
  * ~~https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89089~~
  * ~~https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89117~~

### CODE REVIEW

File: include/seqan3/std/iterator [hunk 1]
```diff
@@ -135,7 +136,7 @@ constexpr auto back_inserter(container_t & container)
  */
 //!\cond
 template <typename t>
-SEQAN3_CONCEPT Readable = static_cast<bool>(::ranges::Readable<t>());
+SEQAN3_CONCEPT Readable = static_cast<bool>(::ranges::Readable<t>);
```
Comment (eseiler) [inline include/seqan3/std/iterator]: We don't need the `static_cast<bool>` anymore, do we?

File: include/seqan3/std/type_traits [hunk 2]
```diff
@@ -14,8 +14,6 @@
 
 #include <type_traits>
 
-#if __cplusplus <= 201810L // TODO feature_test_macro if available
```
Comment (eseiler) [inline include/seqan3/std/type_traits]: So you always "overwrite" the `type_identity` ? Is the one in c++20 different?

File: include/seqan3/std/view/subrange.hpp [hunk 3]
```diff
@@ -66,7 +66,7 @@ namespace seqan3::view
  * \hideinitializer
  */
 template <std::Iterator it_t, std::Sentinel<it_t> sen_t>
-using subrange = std::ranges::iterator_range<it_t, sen_t>;
+using subrange = ::ranges::subrange<it_t, sen_t>;
 //TODO change to ranges::subrange once that has arrived
```
Comment (eseiler) [inline include/seqan3/std/view/subrange.hpp]: remove todo

File: include/seqan3/io/alignment_file/sam_tag_dictionary.hpp [hunk 4]
```diff
@@ -58,13 +59,24 @@ namespace seqan3
...
  * \sa seqan3::sam_tag_dictionary
  */
+
+#ifdef __cpp_nontype_template_parameter_class
+template <constexpr_string<2> str> // TODO: better handling if too large string is provided?
+constexpr uint16_t operator""_tag()
+{
+#else // GCC/Clang extension
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/io/alignment_file/sam_tag_dictionary.hpp]: The code below is an extension in GCC/Clang, GCC9 now warns in pedantic mode that this extension is not canonical c++ and never made it into C++20.

File: test/header/CMakeLists.txt [hunk 5]
```diff
@@ -103,7 +103,9 @@ seqan3_require_test ()
 
 seqan3_header_test (seqan3 "${SEQAN3_CLONE_DIR}/include")
 seqan3_header_test (seqan3_test "${SEQAN3_CLONE_DIR}/test/include")
-seqan3_header_test (range-v3 "${SEQAN3_CLONE_DIR}/submodules/range-v3/include")
+
+# contains deprecated that error on inlusion
```
Comment (smehringer) [inline test/header/CMakeLists.txt]: :beer:

File: include/seqan3/std/ranges [hunk 6]
```diff
@@ -183,10 +185,15 @@ using SEQAN3_DOXYGEN_ONLY(empty =) ::ranges::empty;
...
 /*!\typedef std::ranges::default_sentinel
-* \brief Alias for ranges::default_sentinel. Empty sentinel type for use with iterator types that know the bound of their range.
+* \brief Alias for ranges::default_sentinel. Empty sentinel object for use with iterators that know the bound of their range.
 */
 using SEQAN3_DOXYGEN_ONLY(default_sentinel =) ::ranges::default_sentinel;
 
+/*!\typedef std::ranges::default_sentinel_t
+* \brief Alias for ranges::default_sentinel_t. Type of ranges::default_sentinel.
+*/
+using SEQAN3_DOXYGEN_ONLY(default_sentinel_t =) ::ranges::default_sentinel_t;
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/std/ranges]: Yes and no, I like to keep the file changes together and I think "adapting std" is a good logical unit and changes in our library are then different.

File: include/seqan3/std/type_traits [hunk 7]
```diff
@@ -29,17 +27,21 @@ namespace std
  * \ingroup type_traits
  */
 template <typename t>
-struct type_identity
+struct type_identity;
+
+//!\cond
+template <typename t>
+    requires true
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/std/type_traits]: in presence of a previous definition these lines appear as a specialisation, but in the absence of a previous definition, these appear as the definition.

**Pull request merged**